src/Psalm/Internal/Analyzer/Statements/Expression/Assignment/ArrayAssignmentAnalyzer.php

```diff
@@ -170,7 +170,10 @@ public static function updateArrayType(
         $key_values = [];
 
         if ($current_dim instanceof PhpParser\Node\Scalar\String_) {
-            $key_values[] = new TLiteralString($current_dim->value);
+            $value_type = Type::getAtomicStringFromLiteral($current_dim->value);
+            if ($value_type instanceof TLiteralString) {
```

Comment:
Seems weird to ignore the key when it's not a literal, no?

Reply:
It kinda does, but then the same happens below if some of the keys are not literal (`if (count($string_literals) + count($int_literals) === count($all_atomic_types))`).

The only difference is that in that branch, `$key_values` is either empty or contains all keys, and here we can have partial results. On the other hand, no tests broke, so I'm not really sure if it's that critical.

Reply:
It could be an interesting exercise to run the entire test suite with `max_str_length = 2`